backend/src/api/classes/dto/existing-class-extended.dto.spec.ts

```diff
@@ -12,48 +12,48 @@ describe("ExistingClassExtendedDto", () => {
   };
 
   it("can be constructed", () => {
-    const classDto = plainToInstance(ExistingClassExtendedDto, klass);
+    const classDto = ExistingClassExtendedDto.fromQueryResult(klass);
 
     expect(classDto.id).toEqual(klass.id);
     expect(classDto.name).toEqual(klass.name);
-    expect(classDto.teacherId).toBeUndefined();
+    expect(classDto).not.toHaveProperty("teacherId");
     expect(classDto.teacher).toEqual(klass.teacher);
     expect(classDto.sessions).toEqual([1, 2]);
     expect(classDto.studentCount).toBe(25);
   });
 
   it("handles empty sessions", () => {
-    const dto = plainToInstance(ExistingClassExtendedDto, {
+    const dto = ExistingClassExtendedDto.fromQueryResult({
       ...klass,
       sessions: [],
-    });
+    } as Class);
```

Comment:
can you define the type of `klass` to be `Class`? Then this should not be necessary :)